proto/stat_filter.proto

```diff
@@ -56,4 +58,219 @@ message Dimension {
 message DimensionFilter {
   Dimension dimension = 1;
   string value = 2;
-}
\ No newline at end of file
+}
+
+enum FilterCategory {
+  UNKNOWN_FILTER_CATEGORY = 0;
+  STRING_FILTER_CATEGORY = 1;
+  INT_FILTER_CATEGORY = 2;
+}
+
+enum FilterArgumentCount {
+  UNKNOWN_FILTER_ARGUMENT_COUNT = 0;
+  ONE_FILTER_ARGUMENT_COUNT = 1;
+  MANY_FILTER_ARGUMENT_COUNT = 2;
+}
+
+enum ObjectTypes {
+  NO_OBJECTS = 0;
+  INVOCATION_OBJECTS = 1;
+  EXECUTION_OBJECTS = 2;
+  TARGET_STATUS_OBJECTS = 3;
+}
+
+message FilterTypeOptions {
+  FilterCategory category = 1;
+  string database_column_name = 2;
+  repeated ObjectTypes supported_objects = 3;
+}
+
+message FilterOperandOptions {
+  repeated FilterCategory supported_categories = 1;
+  string database_query_string = 2;
+  FilterArgumentCount argument_count = 3;
+}
+
+extend google.protobuf.EnumValueOptions {
+  FilterTypeOptions filter_type_options = 7000;
+  FilterOperandOptions filter_operand_options = 7001;
+}
+
+message FilterValue {
+  repeated string string_value = 1;
+  repeated int64 int_value = 2;
+}
+
+enum FilterType {
+  UNKNOWN_FILTER_TYPE = 0 [(filter_type_options) = {
+    supported_objects:
+      [NO_OBJECTS];
```

Comment:
nit: could NO_OBJECTS be represented as an empty list instead? (and maybe rename NO_OBJECTS => UNKNOWN_OBJECTS to match the usual UNKNOWN_* convention?)

Reply:
Yeah, this is better, guess I got tunnel vision from how I handled filter categories.